storage/ndb/src/kernel/vm/mt.cpp

```diff
@@ -11840,37 +11891,39 @@ mt_get_threads_for_blocks_no_proxy(const Uint32 blocks[],
 static bool
 may_communicate(unsigned from, unsigned to)
 {
-  if (is_main_thread(from) ||
-      is_main_thread(to))
+  bool can_communicate = false;
+  if (is_main_or_rep_thread(from) ||
+      is_main_or_rep_thread(to))
   {
     // Main threads communicates with all other threads
-    return true;
+    can_communicate = true;
   }
-  else if (is_tc_thread(from))
+  if (is_tc_thread(from))
   {
-    // TC threads can communicate with SPJ-, LQH-, main- and itself
-    return is_ldm_thread(to)  ||
-           is_tc_thread(to);      // Cover both SPJs and itself
+    /**
+     * TC threads can communicate other TC threads, with LDM
+     * threads and with all blocks containing Query blocks,
+     * thus all other threads.
+     */
+    can_communicate = true;
   }
-  else if (is_ldm_thread(from))
+  if (is_ldm_thread(from))
   {
     // All LDM threads can communicates with TC-, main- and ldm.
-    return is_tc_thread(to)   ||
-           is_ldm_thread(to) ||
-           is_recover_thread(to) ||
-           (to == from);
+    can_communicate = can_communicate ||
+                      is_tc_thread(to)   ||
+                      is_ldm_thread(to) ||
+                      (to == from);
   }
-  else if (is_recover_thread(from))
+  if (is_recv_thread(from))
```

Comment:
In the previous implementation, recv thread could not communicate with other recv threads, since we combine threads together, how can we keep this limitation? Here we just loose it.

Reply:
Since the receive threads will always contain Query workers, this means that a recv thread can always send signals to other recv threads, so it is lost because it can no longer happen.